Tests/MongoKittenTests/CRUDTests.swift

```diff
@@ -1,55 +1,180 @@
+    XCTAssertEqual(try schema.count().wait(), startingCount+1)
+  }
+
+  func testBulkCreateDummyAccounts () throws { 
+    struct SetupFailed: Error {}
+    let schema = mongo[DummyAccount.collectionName]
+    let startingCount = try schema.count().wait()
+
+    let dummyAccounts = [
+      DummyAccount(Name: "Test", Password: "ing", Age: 77),
+      DummyAccount(Name: "To", Password: "see", Age: 8),
+      DummyAccount(Name: "If", Password: "bulk", Age: 10),
+      DummyAccount(Name: "Inserts", Password: "will", Age: 15),
+      DummyAccount(Name: "Work", Password: "as", Age: 19),
+      DummyAccount(Name: "I", Password: "expect", Age: 30),
+      DummyAccount(Name: "Them", Password: "to", Age: 82),
+    ]
+    _ = schema.insertManyEncoded(dummyAccounts)
+    XCTAssertEqual(try schema.count().wait(), startingCount+7)
+  }
+
+  func readDummyAccount (name : String) throws -> DummyAccount? {
+    let schema = mongo[DummyAccount.collectionName]
+    if let dummy = try schema.findOne("name" == name, as: DummyAccount.self).wait() {
+      return dummy
+    }
+    return nil
+  }
+
+  func printDummyAccounts () throws {
+    let schema = mongo[DummyAccount.collectionName]
+    try schema.find().forEach{ dummy in 
+      print("\(dummy["_id"]!): Name - \(dummy["name"]!) | Password - \(dummy["password"]!) | Age - \(dummy["age"]!)")
+    }.wait()
+  }
+
+  func testReadDummyAccounts () throws {
+    try testBulkCreateDummyAccounts()
+    if let dummy = try readDummyAccount(name: "Them") {
+        XCTAssertEqual(dummy.password, "to")
+        XCTAssertEqual(dummy.age, 82 )
+    } else {
+      XCTFail("Retrieved a nil value")
+    }
+  }
+
+  func testUpdateDummyAccounts () throws {
+    try testBulkCreateDummyAccounts()
+    if var dummy = try readDummyAccount(name: "Them") {
+      let schema = mongo[DummyAccount.collectionName]
+      dummy.name = "UpdateTest"
+      _ = try schema.updateEncoded(where: "_id" == dummy._id, to: dummy).wait()      
+      let updatedDummy = try readDummyAccount(name: "UpdateTest")
+      XCTAssertNotNil(updatedDummy)
+      XCTAssertEqual(updatedDummy?._id, dummy._id)
+    } else {
+      XCTFail("Retrieved a nil value")
+    }
+  }
+
+  func testDeleteDummyAccounts () throws {
+    try testBulkCreateDummyAccounts()
+
+    if let dummy = try readDummyAccount(name: "Them") {
+      let schema = mongo[DummyAccount.collectionName]
+      _ = try schema.deleteOne(where: "_id" == dummy._id).wait()
+      XCTAssertNil(try readDummyAccount(name: "Them"))
+    } else {
+      XCTFail("Retrieved a nil value")
+    }
+  }
+
+  func testBulkUpdateDummyAccounts() throws {
+    try testBulkCreateDummyAccounts()
+
+    let schema = mongo[DummyAccount.collectionName]
+    _ = try schema.updateMany(where: "age" < 18, to: ["$set" : ["name": "Underaged"]]).wait()
+    schema.find("age" < 18).forEach{dummy in 
+      //XCTAssert(dummy[])
+    }
+  }
+}
+
+// class RemoteDatabaseCRUDTests : XCTestCase {
```

Comment:
You can safely remove this commented code when you're done. This lives in git, so there's no need to clutter the file